src/header.iced

```diff
@@ -14,50 +13,60 @@ crypto_auth_KEYBYTES = 32
 crypto_secretbox_KEYBYTES = 32
 
 compute_mac_key = (encryptor, header_hash, pubkey) ->
-  zero_bytes = Buffer.alloc(crypto_auth_KEYBYTES)
+  zero_bytes = new Buffer(crypto_auth_KEYBYTES)
+  zero_bytes.fill(0)
   mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
   # take last crypto_auth_BYTES bytes of MAC box
   return mac_box.slice(-crypto_auth_KEYBYTES)
 
-exports.generate_encryption_header_packet = (encryptor, recipients) ->
+exports.generate_encryption_header_packet = (encryptor, recipients, opts) ->
+  # create the header list and populate the quick stuff
   header_list = []
   header_list.push('saltpack')
   header_list.push([current_major, current_minor])
   header_list.push(encryption_mode)
 
-  payload_encryptor = crypto.alloc({force_js : true})
-  payload_key = prng(crypto_secretbox_KEYBYTES)
+  # generate the payload and ephemeral keys
+  payload_encryptor = nacl.alloc({force_js : false})
+  payload_key = crypto.randomBytes(crypto_secretbox_KEYBYTES)
   payload_encryptor.secretKey = payload_key
-  ephemeral_encryptor = crypto.alloc({force_js : true})
+  ephemeral_encryptor = nacl.alloc({force_js : false})
   ephemeral_encryptor.genBoxPair()
   header_list.push(ephemeral_encryptor.publicKey)
 
-  sender_sbox = payload_encryptor.secretbox({plaintext : encryptor.publicKey, nonce : nonce.nonceForSenderKeySecretBox()})
+  # support anonymous senders
+  sender_encryptor = if encryptor.publicKey? then encryptor else ephemeral_encryptor
```

Comment:
What exactly is the contract for the caller to specify anonymous encryption? It looks like they must pass in a non-null encryptor that has a null public key. Any reason not to support a null encryptor arg? I don't have a strong opinion one way or the other, but let's document what we decide to do somehow.